src/dsp_tools/commands/ingest_xmlupload/user_information.py

```diff
@@ -0,0 +1,121 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any
+
+import pandas as pd
+
+# pylint: disable=too-few-public-methods
+
+
+separator = "\n    "
+list_separator = "\n    - "
+
+
+@dataclass(frozen=True)
+class IngestInformation:
+    """
+    This class stores the information about the mapping of ids provided by the dsp-ingest service
+    and the filepaths used in the XML file.
+    """
+
+    unused_media_paths: list[str]
+    media_no_id: list[tuple[str, str] | Any]
+    maximum_prints: int = field(default=20)
+    csv_filepath: Path = field(default=Path.cwd())
+    unused_media_filename: str = "UnusedMediaUploadedInSipi.csv"
+    no_id_filename: str = "NotUploadedFilesToSipi.csv"
+
+    def ok_msg(self) -> str | None:
+        """
+        This function checks if no media was unused or not uploaded.
+        If that is the case it returns the message,
+        if not, it ends without an effect.
+
+        Returns:
+            Message if all went well.
+        """
+        if not self.unused_media_paths and not self.media_no_id:
+            return (
+                "All multimedia files referenced in the XML file were uploaded through dsp-ingest.\n"
+                "No multimedia files were uploaded through dsp-ingest that were not referenced in the XML file."
+            )
+        return None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function generates the user message and saves a file with the information
+        if a lot of resources are affected.
+
+        Returns:
+            User message
+        """
+        self._check_csv_if_applicable()
+        return self._get_error_msg()
+
+    def _get_error_msg(self) -> str:
+        msg_list = [
+            "The upload cannot continue as there are problems with the multimedia files referenced in the XML.",
+        ]
+        if has_msg := self._get_unused_path_msg():
+            msg_list.append(has_msg)
+        if has_msg := self._get_no_id_msg():
+            msg_list.append(has_msg)
+        return separator.join(msg_list)
+
+    def _get_no_id_msg(self) -> str | None:
+        if 0 < len(self.media_no_id) <= self.maximum_prints:
+            return (
+                "The data XML file contains references to the following multimedia files "
+                "which were not previously uploaded through dsp-ingest:"
+                + list_separator
+                + list_separator.join([f"Resource ID: '{x[0]}' | Filepath: '{x[1]}'" for x in self.media_no_id])
+            )
+        elif len(self.media_no_id) > self.maximum_prints:
+            return (
+                "The data XML file contains references to multimedia files "
+                "which were not previously uploaded through dsp-ingest:\n"
+                f"    The file with the resource IDs and filenames was saved at "
```

Comment:
https://owl.purdue.edu/owl/general_writing/grammar/using_articles.html

Reply:
My reasoning is: For us devs, the file is known, we refer to it as "the file". But the user hears it the first time, it was just saved. So for the user, it would be more natural to hear that we saved "a" file for them. But that's a detail, so if you prefer "the", that's fine for me :-)